lib/gui/app/components/image-selector/image-selector.jsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2016 resin.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict'
+
+const React = require('react')
+const ReactDOM = require('react-dom')
+const propTypes = require('prop-types')
+
+const middleEllipsis = require('./../../utils/middle-ellipsis')
+
+const { Provider, Button } = require('rendition')
+
+const shared = require('/./../../../../../lib/shared/units')
+const { StepButton, StepNameButton, StepSelection,
+  Footer, Underline, SizeText, ChangeButton } = require('./../../styled-components')
+
+class SelectImageButton extends React.Component {
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      chosenImage: '',
+      imageSize: ''
+    }
+  }
+
+  static getDerivedStateFromProps (props, state) {
+    state.chosenImage = middleEllipsis(props.getImageName || props.getImageBasename , 20)
+    state.imageSize = shared.bytesToClosestUnit(props.getImageSize)
+    return state
+  }
+
+
+  render () {
+    if (this.props.hasImage){
+      return (
+        <Provider>
+          <StepSelection>
+            <StepNameButton
+              plaintext
+              onClick={() => this.props.showSelectedImageDetails()}
```

Comment:
Since this is not tsx, you might even be able to do `onClick={this.props.showSelectedImageDetails}` but keep in mind that the method will obviously be called with an event as its first argument.
We might prefer not doing so though, if there are plans to move to tsx.

Reply:
@thgreasi `onClick={this.props.showSelectedImageDetails}` doesn't work. It calls the `this.props.showSelectedImageDetails` method once only, when component is rendered

Reply:
It sounds like you had
```
onClick={this.props.showSelectedImageDetails()}
```
instead of 
```
onClick={this.props.showSelectedImageDetails}
```

Reply:
Truth <a-little-embarrased>
So: why `onClick={this.props.showSelectedImageDetails}` could be better then `onClick={() => this.props.showSelectedImageDetails}`. Just shorter or more? What does it mean that method will obviously be called with an event as its first argument?
@thundron are there plans to move to tsx? I believe that even if then quite distant?

Reply:
@amdomanska They are distant for now, but not that much, it *should* be a matter of refining the webpack build and start using .ts/x 
The former is indeed shorter and since you're passing the function **body** instead of a function executing your function (e.g. `() => yourMethod()`) and since the first argument of `onClick` is the click `event`, it'll be passed to the function body (e.g. it's like writing (evt) => yourMethod(evt)`).

Reply:
@thundron @thgreasi  but I understand that typescript doesn't understand that syntax? Or why "We might prefer not doing so though, if there are plans to move to tsx."? In that case I'll leave it like it is, right?

Reply:
@amdomanska I don't understand what the issue would be with that syntax using typescript, the only additional thing would be to include typings afaik and that would maybe feel a bit odd if we don't really use the event parameter inside our function

Reply:
@amdomanska Using arrow functions in react props can cause unnecessary renders and slow down your application. Because the function is created everytime the render method happens, React thinks that the component has a new property, so attempts to rerender it. See https://medium.freecodecamp.org/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36 for a nice explanation.